pages/settings/index.vue

```diff
@@ -48,101 +48,90 @@
 <script>
 import gql from 'graphql-tag'
 
-import { mapGetters } from 'vuex'
+import { mapGetters, mapMutations } from 'vuex'
 import { CancelToken } from 'axios'
 import find from 'lodash/find'
 
 let timeout
 const mapboxToken = process.env.MAPBOX_TOKEN
 
+const query = gql`
+  query getUser($id: ID) {
+    User(id: $id) {
+      id
+      name
+      locationName
+      about
+    }
+  }
+`
+
+const mutation = gql`
+  mutation($id: ID!, $name: String, $locationName: String, $about: String) {
+    UpdateUser(
+      id: $id
+      name: $name
+      locationName: $locationName
+      about: $about
+    ) {
+      id
+      name
+      locationName
+      about
+    }
+  }
+`
+
 export default {
   data() {
     return {
       axiosSource: null,
       cities: [],
       sending: false,
-      form: {
-        name: null,
-        locationName: null,
-        about: null
-      }
+      formData: {}
     }
   },
   computed: {
     ...mapGetters({
-      user: 'auth/user'
-    })
-  },
-  watch: {
-    user: {
-      immediate: true,
-      handler: function(user) {
-        this.form = {
-          name: user.name,
-          locationName: user.locationName,
-          about: user.about
-        }
+      currentUser: 'auth/user'
+    }),
+    form: {
+      get: function() {
+        const { name, locationName, about } = this.currentUser
+        return { name, locationName, about }
+      },
+      set: function(formData) {
+        this.formData = formData
```

Comment:
@tirokk your suggestion yesterday was right: Having a temporary `formData` object solved the issues for me. All that we had to do is to remember to send `formData` not `form` as the variables in the mutation to the backend.